vimiv/api/settings.py

```diff
@@ -315,8 +316,51 @@ def __str__(self) -> str:
         return "String"
 
 
-# Initialize all settings
+class OrderSetting(Setting):
+    """Stores an ordering setting."""
+
+    typ = str
+
+    ORDER_TYPES: Dict[str, Tuple[Callable[..., Any], bool]] = {
+        "alphabetical": (os.path.basename, False),
+        "alphabetical-desc": (os.path.basename, True),
+        "natural": (natural_sort, False),
+        "natural-desc": (natural_sort, True),
+        "recently-modify-first": (os.path.getmtime, False),
+        "recently-modify-last": (os.path.getmtime, True),
+    }
+
+    def __init__(
+        self,
+        *args: Any,
+        additional_order_types: Dict[str, Tuple[Callable[..., Any], bool]] = None,
+        **kwargs: Any,
+    ):
+        super().__init__(*args, **kwargs)
+        self.order_types = dict(self.ORDER_TYPES)
+        if additional_order_types:
+            self.order_types.update(additional_order_types)
+
+    def convert(self, value: str) -> str:
+        if value not in self.order_types:
+            raise ValueError(f"Option must be one of {', '.join(self.ORDER_TYPES)}")
+        return value
+
+    def get_para(self) -> Tuple[Callable[..., Any], bool]:
+        """Returns the ordering parameters according to the current set value."""
+        try:
+            return self.order_types[self.value]
+        except KeyError:
+            raise ValueError(f"Option must be one of {', '.join(self.ORDER_TYPES)}")
 
+    def suggestions(self) -> List[str]:
+        return [str(value) for value in self.order_types]
```

Comment:
Mini nit-pick, `list(self.order_types)` should suffice here, all keys are strings anyway, no?

Reply:
Indeed true. That is much cleaner